extensions/ql-vscode/src/extension.ts

```diff
@@ -909,7 +909,15 @@ async function activateWithInstalledDistribution(
 
   ctx.subscriptions.push(
     commandRunner('codeQL.mockVariantAnalysisView', async () => {
-      const variantAnalysisView = new VariantAnalysisView(ctx);
+      const variantAnalysisView = new VariantAnalysisView(ctx, 1);
+      variantAnalysisView.openView();
+    })
+  );
+
+  // The "openVariantAnalysisView" command is internal-only.
+  ctx.subscriptions.push(
+    commandRunner('codeQL.openVariantAnalysisView', async (variantAnalysisId: number) => {
+      const variantAnalysisView = new VariantAnalysisView(ctx, variantAnalysisId);
```

Comment:
What's the reason for having an internal-only command?

Reply:
This mostly just follows the existing examples in the extension. For example, `codeQL.copyRepoList` and `codeQL.autoDownloadRemoteQueryResults` are both internal-only commands. I would suspect these were implemented as such so they can be easily called from anywhere in the extension.

Reply:
Cool, thanks for explaining 👍